client/my-sites/themes/themes-banner/random-themes-banner.jsx

```diff
@@ -21,13 +17,29 @@ class RandomThemesBanner extends PureComponent {
 		const { banners } = props;
 		this.state = {
 			BannerComponent: sample( banners ),
+			isClient: false,
 		};
 	}
 
+	componentDidMount() {
+		// In this particular case, we need to disable the ESLint check
+		// to disable using `setState()` on `componentDidMount()`, since
+		// it needs to be done this way to properly render the component
+		// both on client-side and server-side.
```

Comment:
Maybe refer to reconciliation errors a bit more directly? Something like

Since we're using randomness to determine which banner to display, we cannot easily guarantee that the client will have the same banner as the server, thus possibly leading to reconciliation errors. To avoid them, we just render `null` on component mount, and only afterwards, we update component state to cause a re-render on the client in order to actually render the component. This means however that we have to `setState` in `componentDidMount`.